cores/esp32/esp32-hal-rmt.c

```diff
@@ -147,26 +192,131 @@ static xSemaphoreHandle g_rmt_block_lock = NULL;
 /**
  * Internal method (private) declarations
  */
-static void _initPin(int pin, int channel, bool tx_not_rx);
 
-static bool _rmtSendOnce(rmt_obj_t* rmt, rmt_data_t* data, size_t size, bool continuous);
+static rmt_obj_t* _rmtAllocate(int pin, int from, int size)
+{
+    size_t i;
+    // setup how many buffers shall we use
+    g_rmt_objects[from].buffers = size;
 
-static void ARDUINO_ISR_ATTR _rmt_isr(void* arg);
+    for (i=0; i<size; i++) {
+        // mark the block of channels as used
+        g_rmt_objects[i+from].allocated = true;
+    }
+    return &(g_rmt_objects[from]);
+}
 
-static rmt_obj_t* _rmtAllocate(int pin, int from, int size);
+static char tag[] = "rmt_receiver";
 
-static void _initPin(int pin, int channel, bool tx_not_rx);
+static void _rmtDumpStatus(rmt_channel_t channel) 
+{
+   bool loop_en;
+   uint8_t div_cnt;
+   uint8_t memNum;
+   bool lowPowerMode;
+   rmt_mem_owner_t owner;
+   uint16_t idleThreshold;
+   uint32_t status;
+   rmt_source_clk_t srcClk;
+
+RMT_MUTEX_LOCK(channel);
+   rmt_get_tx_loop_mode(channel, &loop_en);
+   rmt_get_clk_div(channel, &div_cnt);
+   rmt_get_mem_block_num(channel, &memNum);
+   rmt_get_mem_pd(channel, &lowPowerMode);
+   rmt_get_memory_owner(channel, &owner);
+   rmt_get_rx_idle_thresh(channel, &idleThreshold);
+   rmt_get_status(channel, &status);
+   rmt_get_source_clk(channel, &srcClk);
+
+   ESP_LOGD(tag, "Status for RMT channel %d", channel);
```

Comment:
do not use ESP_LOGx in Arduino. Use log_x

Reply:
OK. Done.